TT-11522 DX-1280 Setup MDCB control plane with Helm Chart (#4491)
TykTechnologies/tyk-docs: tyk-docs/content/tyk-multi-data-centre/setup-controller-data-centre.md

## Introduction
The [Tyk Control Plane]({{< ref "tyk-multi-data-centre/mdcb-components.md#control-plane" >}}) will contain all the standard components of a standard Self-Managed installation with the addition of one additional component: the Multi Data Centre Bridge (MDCB).

> ## Installing MDCB Component with package

Reviewer: Is this a Link RPM, Debian package? Should the section be named: _Installing MDCB Component On Linux_ and then have subsections for installing using RPM and Debian packages?

Author: Updated

The MDCB component must be able to connect to Redis and MongoDB/PostgreSQL directly from within the Control Plane deployment. It does not require access to the Tyk Gateway(s) or Dashboard application.

> The MDCB component will however, by default, expose an RPC service on port 9091, to which the Tyk Data Plane (Worker gateway(s)) data centres will need connectivity.

Reviewer: Add an brief sentence or two to explain Tyk Data Plane, similar to what has been done for Tyk Control Plane in introduction

Author: Added

```curl
sudo yum install tyk-sink
```

## Installing in a Kubernetes Cluster with our Helm Chart

The [Tyk Control Plane]({{<ref "product-stack/tyk-charts/tyk-control-plane-chart">}}) helm chart is pre-configured to install Tyk control plane

> for multi data centre API management from a single Dashboard with the MDCB component.

Reviewer: remove line break

### Prerequisites

> Before installation, you need to prepare the followings:

Reviewer: Remove sentence and list pre-requisites

- [Kubernetes 1.19+](https://kubernetes.io/docs/setup/)
- [Helm 3+](https://helm.sh/docs/intro/install/)
- MDCB and Dashboard license

> ### Quick Start

Reviewer: Convert number steps to numbered section headings #### 1. Step 1, #### 2. Step 2 etc.

Author: Updated

Ensure that you are installing PostgreSQL versions that are supported by Tyk. Please consult the list of [supported versions]({{< ref "tyk-dashboard/database-options" >}}) that are compatible with Tyk.
{{< /note >}}

**4. Install Tyk Control Plane**
```bash
helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/

helm repo update

helm upgrade tyk-cp tyk-helm/tyk-control-plane -n $NAMESPACE \
  --install \
  --set global.adminUser.useSecretName=admin-secrets \
  --set global.secrets.useSecretName=my-secrets \
  --set tyk-mdcb.mdcb.useSecretName=mdcb-secrets \
  --set global.redis.addrs="{tyk-redis-master.$NAMESPACE.svc:6379}" \
  --set global.redis.passSecret.name=tyk-redis \
  --set global.redis.passSecret.keyName=redis-password \
  --set global.postgres.connectionStringSecret.name=postgres-secrets \
  --set global.postgres.connectionStringSecret.keyName=postgresUrl
```

**5. Done!**

Now Tyk Dashboard and Tyk MDCB should be accessible through service `dashboard-svc-tyk-control-plane-tyk-dashboard` at port `3000` and `mdcb-svc-tyk-control-plane-tyk-mdcb` at port `9091` respectively. You can login to Dashboard using the admin email and password to start managing APIs.

> You can get MDCB connection details from the installation output, to prepare installation of [setup MDCB Data Plane]({{<ref "product-stack/tyk-charts/tyk-data-plane-chart">}}):

Reviewer: ```suggestion
You can use the MDCB connection details included in the installation output, to install the [MDCB Data Plane]({{<ref "product-stack/tyk-charts/tyk-data-plane-chart">}}):
```

- MDCB and Dashboard license

### Quick Start

#### Step 1 - Setup required credentials

First, you need to provide Tyk Dashboard and MDCB license, admin email and password, and API keys. We recommend to store them in secrets.

```bash
NAMESPACE=tyk-cp

API_SECRET=changeit
ADMIN_KEY=changeit
ADMIN_EMAIL=admin@default.com
ADMIN_PASSWORD=changeit
DASHBOARD_LICENSE=changeit
MDCB_LICENSE=changeit

kubectl create namespace $NAMESPACE

kubectl create secret generic my-secrets -n $NAMESPACE \
    --from-literal=APISecret=$API_SECRET \
    --from-literal=AdminSecret=$ADMIN_KEY \
    --from-literal=DashLicense=$DASHBOARD_LICENSE

kubectl create secret generic mdcb-secrets -n $NAMESPACE \
    --from-literal=MDCBLicense=$MDCB_LICENSE

kubectl create secret generic admin-secrets -n $NAMESPACE \
    --from-literal=adminUserFirstName=Admin \
    --from-literal=adminUserLastName=User \
    --from-literal=adminUserEmail=$ADMIN_EMAIL \
    --from-literal=adminUserPassword=$ADMIN_PASSWORD
```

#### Step 2 - Install Redis (if you don't already have Redis installed)

If you do not already have Redis installed, you may use these charts provided by Bitnami.

```bash

> helm upgrade tyk-redis oci://registry-1.docker.io/bitnamicharts/redis -n $NAMESPACE --install --version 19.0.2

Reviewer 2: Should we also add a note to install supported redis version like you have done for postgres?

Author: Added

```

> Follow the notes from the installation output to get connection details and password. The DNS name of your Redis as set by Bitnami is `tyk-redis-master.tyk.svc:6379` (Tyk needs the name including the port) 

Reviewer 2: Since you are setting namespace as `tyk-cp`
```suggestion
Follow the notes from the installation output to get connection details and password. The DNS name of your Redis as set by Bitnami is `tyk-redis-master.tyk-cp.svc:6379` (Tyk needs the name including the port) 
```

{{< /note >}}

#### Step 4 - Install Tyk Control Plane
```bash
helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/

helm repo update

helm upgrade tyk-cp tyk-helm/tyk-control-plane -n $NAMESPACE \
  --install \
  --set global.adminUser.useSecretName=admin-secrets \
  --set global.secrets.useSecretName=my-secrets \
  --set tyk-mdcb.mdcb.useSecretName=mdcb-secrets \
  --set global.redis.addrs="{tyk-redis-master.$NAMESPACE.svc:6379}" \
  --set global.redis.passSecret.name=tyk-redis \
  --set global.redis.passSecret.keyName=redis-password \
  --set global.postgres.connectionStringSecret.name=postgres-secrets \
  --set global.postgres.connectionStringSecret.keyName=postgresUrl
```

#### Step 5 - Done!

Now Tyk Dashboard and Tyk MDCB should be accessible through service `dashboard-svc-tyk-cp-tyk-dashboard` at port `3000` and `mdcb-svc-tyk-cp-tyk-mdcb` at port `9091` respectively. You can login to Dashboard using the admin email and password to start managing APIs.

You can use the MDCB connection details included in the installation output, to install the [MDCB Data Plane]({{<ref "tyk-multi-data-centre/setup-worker-data-centres">}}).

> For the complete installation guide and configuration options, please see [Tyk Control Plane Helm Chart]({{<ref "product-stack/tyk-charts/tyk-control-plane-chart">}}).

Reviewer 3: Just an opinion, maybe we can mention the complete installation guide beforehand, right under the `## Installing in a Kubernetes Cluster with our Helm Chart` What do you think?

Author: Thanks, updated as suggested
